client/src/main/java/com/kritsit/casetracker/client/domain/ui/controller/AdministratorController.java

```diff
@@ -0,0 +1,134 @@
+package com.kritsit.casetracker.client.domain.ui.controller;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
+import com.kritsit.casetracker.client.domain.services.IAdministratorService;
+import com.kritsit.casetracker.client.domain.services.ServerConnection;
+import com.kritsit.casetracker.shared.domain.Request;
+import com.kritsit.casetracker.shared.domain.model.Permission;
+import com.kritsit.casetracker.shared.domain.model.Staff;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.TableView;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TextField;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.stage.Stage;
+import javafx.fxml.FXML;
+
+public class AdministratorController implements IController {
+    
+    private ObservableList<Staff> staffList;
+    private IAdministratorService administratorService;
+    private Stage stage;
+    private final Logger logger = LoggerFactory.getLogger(AdministratorController.class);
+    
+    public void setAdministratorService(IAdministratorService administratorService) {
+        this.administratorService = administratorService;
+        
+    }
+    
+    public void initFrame(){
+        logger.info("Initiating frame");
+        if (administratorService.getUser().getPermission() == Permission.ADMIN) {
+            initialize();
+            initStaffTable();
+            initPermissionCombobox();
+        } else {
+            logger.debug("Administrator view disabled");
+        } 
+    }
+    
+    
+    public void initialize(){
+        
+        addUserButton.setOnAction(event->{
+            Permission permission = Permission.valueOf(permissionCombobox.getValue());
+            
+            Staff staff = new Staff(usernameField.getText(), 
+                    firstNameField.getText(), 
+                    lastNameField.getText(), 
+                    departmentCombobox.getValue(), 
+                    positionField.getText(), 
+                    permission);
+            
+            List<Staff> list = new ArrayList<Staff>();
+            list.add(staff);
+            Request request = new Request("addUser", list);
+           
+            // sernding serialized request through socket
```

Comment:
This would all be done in the Administrator class (validation and verification) and the ServerConnection class (actually sending the request to the server and interpreting the result)